dask_cuda/utils.py

```diff
@@ -181,12 +227,17 @@ def get_n_gpus():
 
 def get_device_total_memory(index=0):
     """
-    Return total memory of CUDA device with index
+    Return total memory of CUDA device with index or with device identifier UUID
     """
     pynvml.nvmlInit()
-    return pynvml.nvmlDeviceGetMemoryInfo(
-        pynvml.nvmlDeviceGetHandleByIndex(index)
-    ).total
+
+    if index and not str(index).isnumeric():
+        # This means index is UUID. This works for both MIG and non-MIG device UUIDs.
+        handle = pynvml.nvmlDeviceGetHandleByUUID(str.encode(str(index)))
```

Comment:
@rjzamora we discussed some time ago that NVML doesn't respect `CUDA_VISIBLE_DEVICES`, and while that's true, one of the most user-friendly ways to get a MIG device handle is by its UUID. It seems that users would prefer something simpler such as `MIG-0-1` (meaning GPU 0, MIG instance 1, for example), instead of the UUID that looks like `MIG-84fd49f2-48ad-50e8-9f2e-3bf0dfd47ccb`. Do you think this is worth bringing up with the NVML (and potentially CUDA) teams?